src/uct/tcp/tcp_net.c

```diff
@@ -59,15 +61,15 @@ ucs_status_t uct_tcp_netif_caps(const char *if_name, double *latency_p,
         speed_mbps = edata.speed;
 #endif
 #if HAVE_DECL_SPEED_UNKNOWN
-        if (speed_mbps == SPEED_UNKNOWN) {
+        speed_known = speed_mbps != SPEED_UNKNOWN;
 #else
-        if ((speed_mbps == 0) || ((uint16_t)speed_mbps == (uint16_t)-1)) {
+        speed_known = (speed_mbps != 0) && ((uint16_t)speed_mbps != (uint16_t)-1);
```

Comment:
minor: why not use uint16_t speed_mbps ?

Reply:
because for older glibc, ethtool_cmd.speed is u16. for newer glibc, we can use SPEED_UNKNOWN constant, and also ethtool_cmd_speed() returns u32